jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/CDataXMLStreamWriter.java

```diff
@@ -0,0 +1,58 @@
+/* 
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.workitem.webservice;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import org.apache.cxf.staxutils.DelegatingXMLStreamWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CDataXMLStreamWriter extends DelegatingXMLStreamWriter {
+
+    private static Logger logger = LoggerFactory.getLogger(CDataXMLStreamWriter.class);
+    private List<String> cdataElements = new ArrayList<>();
+    private String currentElementName;
+
+    public CDataXMLStreamWriter(XMLStreamWriter del, String cdataElementStr) {
+        super(del);
+        this.cdataElements = Stream.of(cdataElementStr.split(",", -1)).map(String::trim).collect(Collectors.toList());;
+    }
+
+    @Override
+    public void writeStartElement(String prefix, String local, String uri) throws XMLStreamException {
+        currentElementName = local; 
+        super.writeStartElement(prefix, local, uri); 
+    } 
+
+    @Override
+    public void writeCharacters(char[] arg0, int arg1, int arg2) throws XMLStreamException {
+        if (cdataElements.contains(currentElementName)) { 
+          String str = new String(arg0);
```

Comment:
You can seve this variable and create the string when calling writeCData

Reply:
done